src/Navigation/SidePanel/common.tsx

```diff
@@ -26,18 +27,29 @@ export const ItemWrapper = styled('div')`
   }
 `
 
-export const Panel = styled('div')`
+export const DefaultPanel = styled.div<SidePanelProps>`
   position: absolute;
+  box-sizing: border-box;
+  width: ${({ width }) => width || '300px'};
+  height: 100lvh;
+  left: 0px;
+  top: 0px;
+
   display: flex;
   flex-direction: column;
   gap: 0.2em;
+
   padding: 5px 25px;
-  width: ${({ style: { width } }: any) => width || 300}px;
-  height: ${viewPortH}px;
-  left: ${({ style: { isOpen, width } }: any) =>
-    isOpen ? '0px' : `-${width + 50}px`};
-  top: 0px;
-  box-shadow: ${({ style: { variant } }: any) =>
-    variant === 'default' ? '0px 4px 5px rgba(0, 0, 0, 0.1)' : 'none'};
-  transition: all 0.7s ease-in-out;
+  transform: translateX(
+    ${({ isOpen, width }) => (isOpen ? '0px' : `-${width || '300px'}`)}
+  );
+  transition: transform 0.7s ease-in-out;
+
+  box-shadow: 0px 4px 5px rgba(0, 0, 0, 0.1);
+`
+
+export const HiiVariantPanel = styled(DefaultPanel)<SidePanelProps>`
```

Comment:
yea, this is where I struggled, but then learned that I can pass Types along styled-components